### ISSUE TITLE
Systemd fails to start latest version of telegraf due to incorrect parameters

### ISSUE THREAD

**Issue by mdechiaro** [puppetlabs/puppet_operational_dashboards#107]
## Describe the Bug

The latest version of telegraf causes systemd to fail to start service with following errors. The module expects to use latest version of telegraf, so it will update it and then systemd fails to start service.

```
2022-11-03 00:05:51 -0500 /Stage[main]/Telegraf::Install/Package[telegraf]/ensure (notice): ensure changed '1.24.2-1' to '1.24.3-1' (corrective)
2022-11-03 00:05:53 -0500 Puppet (err): Systemd start for telegraf failed!
journalctl log for telegraf:
Nov 03 00:05:38 hostname.domain.com systemd[1]: Stopping The plugin-driven server agent for reporting metrics into InfluxDB...
Nov 03 00:05:43 hostname.domain.com systemd[1]: telegraf.service: Deactivated successfully.
Nov 03 00:05:43 hostname.domain.com systemd[1]: Stopped The plugin-driven server agent for reporting metrics into InfluxDB.
Nov 03 00:05:43 hostname.domain.com systemd[1]: telegraf.service: Consumed 1h 54min 27.931s CPU time.
Nov 03 00:05:46 hostname.domain.com systemd[1]: Starting Telegraf...
Nov 03 00:05:47 hostname.domain.com telegraf[433407]: 2022-11-03T05:05:47Z W! DeprecationWarning: Option "Databases" of plugin "inputs.postgresql_extensible" deprecated since version
1.22.4 and will be removed in 2.0.0: use the sqlquery option to specify database to use
Nov 03 00:05:47 hostname.domain.com telegraf[433407]: 2022-11-03T05:05:47Z W! DeprecationWarning: Option "Databases" of plugin "inputs.postgresql_extensible" deprecated since version
1.22.4 and will be removed in 2.0.0: use the sqlquery option to specify database to use
Nov 03 00:05:47 hostname.domain.com telegraf[433407]: 2022-11-03T05:05:47Z E! [telegraf] Error running agent: error loading config file
/etc/telegraf/telegraf.d/puppetdb_metrics.conf: plugin inputs.http: line 1: configuration specified the fields ["timestamp_key"], but they weren't used
Nov 03 00:05:47 hostname.domain.com systemd[1]: telegraf.service: Main process exited, code=exited, status=1/FAILURE
Nov 03 00:05:47 hostname.domain.com systemd[1]: telegraf.service: Failed with result 'exit-code'.
Nov 03 00:05:47 hostname.domain.com systemd[1]: Failed to start Telegraf.
Nov 03 00:05:47 hostname.domain.com systemd[1]: telegraf.service: Consumed 1.370s CPU time.
Nov 03 00:05:48 hostname.domain.com systemd[1]: telegraf.service: Scheduled restart job, restart counter is at 1.
Nov 03 00:05:48 hostname.domain.com systemd[1]: Stopped Telegraf.
```

## Expected Behavior

Expected behavior is to have module configure telegraf so systemd start it without issue.

## Steps to Reproduce

Since Ubuntu is forced to use latest telegraf release, then this issue will occur repeatedly.

## Environment
os: ubuntu 22.04
puppet-agent: 7.18.0-1jammy
puppetlabs-puppet_operational_dashboards: 1.7.0
puppet-telegraf: 4.3.0

## Additional Context
Unfortunately, we can't ensure a specific version in module because it expects to use the latest.

```
---
puppet_operational_dashboards::telegraf::agent::version: 1.24.2-1
```

```
Warning: /Stage[main]/Puppet_operational_dashboards::Telegraf::Agent/Notify[telegraf_ubuntu_warn]/message: current_value 'absent', should be 'Using \'latest\' for Telegraf package on Ubuntu'
(noop)
```

As a workaround, the `apt` module can be used to pin version. This allows telegraf and Puppet to run without error.

```
---
apt::pins:
  hold-telegraf:
    packages: telegraf
    version: 1.24.2-1
    priority: 1001
```

**Comment by m0dular** [comment]
Thanks for the bug report.  We had seen that only the latest version of the package was available in the Ubuntu repos, but it looks like 1.23 and 1.24 are currently available https://repos.influxdata.com/ubuntu/packages/

We should probably stop enforcing `latest` and remove this warning, I'll look into it https://github.com/puppetlabs/puppet_operational_dashboards/blob/main/manifests/telegraf/agent.pp#L150-L155

**Comment by m0dular** [comment]
Should be fixed by #109 in version 1.8.0, let us know if you see the issue reoccur.
